### ISSUE TITLE
508-defect-2 Fast Follow! [COGNITION]: Headings MUST only be read once

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#14482]
# [508-defect-2 Fast Follow!](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

cc @1Copenut — I'm on the fence if this is a 508-defect-1 or 2. Hearing the heading repeat is a broken experience, but it's easy to work past, so I made this a fast follow. Does that sound right?

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Point of Contact

**VFS Point of Contact:** Jennifer

## User Story or Problem Statement

As a VoiceOver screen reader user, I expect to read the content of a page without unexpected duplication. 

## Details

Headings must only be read once by a screen reader. 

On select pages, those with "Related forms and instructions", Voiceover will read a heading twice. Upon investigation, it is due to the `h4` having a `span` inside it. I tried changing the `span` to a `dfn`, adding `role="text"` (as suggested in [this blog post on text splitting](https://axesslab.com/text-splitting/)), but neither worked. Changing the `h4` to a `p` fixed the issue. 

### Sample pages where this occurs

- https://www.va.gov/find-forms/about-form-21p-0969/

- https://www.va.gov/find-forms/about-form-10-7959c/

## Acceptance Criteria

- [ ] Headings should only read once in screen readers

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Environment

* Operating System: MacOS
* Browser: Safari Version 14.0 (14610.1.28.1.9)
* Screenreading device: VoiceOver
* Server destination: production

## Steps to Recreate

1. Enter `https://www.va.gov/find-forms/about-form-21p-0969/` in browser
2. Start screenreading device listed in Environment
3. Navigate to the heading, "Related forms and instructions"
4. Listen to the h3 "VA Form 21P-534" and then the h4 "Application for Dependency and Indemnity Compensation, Survivors Pension and Accrued Benefits by a Surviving Spouse or Child (Including Death Compensation if Applicable)"
5. Verify that the screen reader says two items, and then repeats the entire heading

### Current code

```
<li>
  <hgroup>
    <h3 id="va-form-21p-534">VA Form 21P-534</h3>
    <h4 class="vads-u-margin-top--0">
      <span class="vads-u-visibility--screen-reader">Form name:</span> Application for Dependency and Indemnity Compensation, Survivors Pension and Accrued Benefits by a Surviving Spouse or Child (Including Death Compensation if Applicable)
    </h4>
  </hgroup>
 <!-- etcetera -->
```
![Screen Shot 2020-10-06 at 6 17 11 PM](https://user-images.githubusercontent.com/57469/95265919-2d6be500-0800-11eb-9fa6-a4f34a8af953.png)

### Recommended solution

Note: I didn't bold the styling, and it was easier to read than the bold text. The designer(s) may prefer a bold h4 styling.

```diff
<li>
  <h3 id="va-form-21p-534">VA Form 21P-534</h3>
+  <p class="vads-u-margin-top--0 vads-u-font-size--h4">
    <dfn class="vads-u-visibility--screen-reader">Form name:</dfn> 
    Application for Dependency and Indemnity Compensation, Survivors Pension and Accrued Benefits by a Surviving Spouse or Child (Including Death Compensation if Applicable) 
  </p>

<!-- etcetera -->
```

![Screen Shot 2020-10-06 at 6 20 44 PM](https://user-images.githubusercontent.com/57469/95266170-acf9b400-0800-11eb-8b83-635b25754028.png)

**Comment by MarciMcGuireGCIO** [comment]
Removing public websites tag since this now falls under the Decision Tools team

**Comment by jenstrickland** [comment]
@zacharymorel Here's a screenshot of what VoiceOver is reading, showing that it reads as "2 items":
 ![Screen Shot 2020-11-19 at 1 48 21 PM](https://user-images.githubusercontent.com/57469/99709939-f11ddc80-2a6d-11eb-843e-9b7ad9ad9c57.png)

Additionally, it then reads the form name a second time, as [shown in this screen recording](https://dsva.slack.com/files/UMLC20J79/F01EY9KTWBG/h4-2items-repeated.mov).

**Comment by zacharymorel** [comment]
@rtwell Tagging you here to get your weigh in. 

The solution that Jenn purposed in the User story works. If we use change that `h3` to a `p` tag and use a definition element for the hidden **Form name** text, the screen reader does not register that theres two elements and does not repeat the "Form Name" + heading text.  

If we bold the `p` tag and give the font size of `h4`, it goes against the heading nesting suggested for a11y by w3, but we keep the current UI look.
Do you have opinions on this?

**Comment by jenstrickland** [comment]
The proper solution is if we retain the `<h4>` semantics and don't hide the "Form name: ". This will prevent VoiceOver from repeating the heading, while retaining the semantic relationship of the `<h3>` and `<h4>` for both sighted and non-sighted users.

What do you think, @rtwell ?

**Comment by rtwell** [comment]
Oh, well if we can make this better for all with this change, I fully support it!

**Comment by zacharymorel** [comment]
Good afternoon  @jenstrickland this is ready to be validated on https://staging.va.gov/ .

**Comment by jenstrickland** [comment]
@zacharymorel Apologies, I think there was some confusion.

### Current staging code

```html
<p class="vads-u-margin-top--0 vads-u-font-size--h4 vads-u-font-weight--bold">
  <dfn class="vads-u-visibility--screen-reader">Form name:</dfn> Application for Dependency and Indemnity Compensation, Survivors Pension and Accrued Benefits by a Surviving Spouse or Child (Including Death Compensation if Applicable)
</p>
```

### Recommended code from the [comment above](https://github.com/department-of-veterans-affairs/va.gov-team/issues/14482#issuecomment-731248366)

Ryan and I discussed and landed on using the h4 and surfacing the "Form name:" for all users. Headings with spans and other elements within tend to cause VoiceOver to announce a number of items and repeat the heading.

```html
<h4 class="vads-u-margin-top--0">
  Form name: Application for Dependency and Indemnity Compensation, Survivors Pension and Accrued Benefits by a Surviving Spouse or Child (Including Death Compensation if Applicable)
</p>
```

**Comment by zacharymorel** [comment]
Whoops. @jenstrickland thanks for clearing that up after the feed back. That can be switched around.

**Comment by zacharymorel** [comment]
@jenstrickland , this is ready again for validation: https://staging.va.gov/find-forms/about-form-21p-0969/

**Comment by jenstrickland** [comment]
Thanks, @zacharymorel! Nice use of `<hgroup>`, too! Although it's been deprecated, I think it's a tidy element, and allows for some handy hooks when needed. Certainly doesn't hurt anything, and makes the markup clearer. 👏  Validated! 🎉
